src/lib.rs

```diff
@@ -390,58 +409,65 @@ pub trait Trait: system::Trait + Sized {
+    trait Store for Module<T: Trait> as Forum {
+
+        /// Map category identifier to corresponding category.
+        pub CategoryById get(category_by_id) config(): map CategoryId => Category<T::BlockNumber, T::Moment, T::AccountId>;
+
+        /// Category identifier value to be used for the next Category created.
+        pub NextCategoryId get(next_category_id) config(): CategoryId;
+
+        /// Map thread identifier to corresponding thread.
+        pub ThreadById get(thread_by_id): map ThreadId => Thread<T::BlockNumber, T::Moment, T::AccountId>;
+
+        /// Thread identifier value to be used for next Thread in threadById.
+        pub NextThreadId get(next_thread_id) config(): ThreadId;
 
-        // A map that has enumerable entries.
-        Bar get(bar) config(): linked_map T::AccountId => BalanceOf<T>;
+        /// Map post identifier to corresponding post.
+        pub PostById get(post_by_id): map PostId => Post<T::BlockNumber, T::Moment, T::AccountId>;
 
-        // this one uses the default, we'll demonstrate the usage of 'mutate' API.
-        Foo get(foo) config(): BalanceOf<T>;
+        /// Post identifier value to be used for for next post created.
+        pub NextPostId get(next_post_id) config(): PostId;
 
+        /// Account of forum sudo.
+        pub ForumSudo get(forum_sudo) config(): T::AccountId;
 
-        // ==== Constraints ====
+        // === Add constraints here ===
 
     }
 }
 
 decl_event!(
-    /// Events are a simple means of reporting specific conditions and
-    /// circumstances that have happened that users, Dapps and/or chain explorers would find
-    /// interesting and otherwise difficult to detect.
     pub enum Event<T>
     where
-        B = BalanceOf<T>,
+        <T as system::Trait>::AccountId,
     {
-        // Just a normal `enum`, here's a dummy event to ensure it compiles.
-        /// Dummy event, just here so there's a generic type that's used.
-        Dummy(B),
+        /// A category was introduced
+        CategoryCreated(CategoryId),
+
+        /// A category with given id was updated. 
+        /// The second argument reflects the new archival status of the category, if changed.
+        /// The third argument reflects the new deletion status of the category, if changed.
+        CategoryUpdated(CategoryId, Option<bool>, Option<bool>),
+
+        /// A thread with given id was created.
+        ThreadCreated(ThreadId),
+
+        /// A thread with given id was moderated.
+        ThreadModerated(ThreadId),
+
+        /// Post with given id was created.
+        PostAdded(PostId),
+
+        /// Post with givne id was moderated.
+        PostModerated(PostId),
+
+        /// Post with given id had its text updated.
+        /// The second argument reflects the number of total edits when the text update occurs.
+        PostTextUpdated(PostId, u64),
+
+        /// Given account was set as forum sudo.
+        ForumSudoSet(AccountId),
```

Comment:
Consider adding `who` set new forum sudo. Could be useful for profiling in case of an attack/bad actor.
```suggestion
        ForumSudoSet(AccountId, AccountId),
```

Reply:
See above.